src/components/ApiAction.vue

```diff
@@ -0,0 +1,416 @@
+  components: {
+    Multipane,
+    MultipaneResizer,
+    QueryCard,
+    SaveQueryModal,
+    QueryList
+  },
+  data() {
+    return {
+      tabs: [],
+      currentTabIdx: 0,
+      showAlert: true,
+      api: null,
+      openapi: null,
+      loading: true,
+      savedQueries: [],
+      newSaveTabIdx: null
+    }
+  },
+  computed: {
+    ...mapGetters('kuzzle', ['$kuzzle', 'currentEnvironment']),
+    ...mapGetters('auth', ['canGetPublicApi', 'canGetOpenApi']),
+    controllers() {
+      return this.api ? Object.keys(this.api) : []
+    },
+    actions() {
+      if (!this.tabs[this.currentTabIdx] || !this.api) {
+        return []
+      }
+      const currentController = this.tabs[this.currentTabIdx].query.controller
+      return currentController && this.api[currentController]
+        ? Object.keys(this.api[currentController])
+        : []
+    },
+
+    emptyTab() {
+      return {
+        query: {
+          controller: null,
+          action: null,
+          body: {}
+        },
+        saved: false,
+        name: '',
+        response: '',
+        savedIdx: null
+      }
+    },
+    currentQueryBody() {
+      if (!this.tabs[this.currentTabIdx]) {
+        return null
+      }
+      return this.tabs[this.currentTabIdx].body
+    },
+    currentQueryName() {
+      if (!this.tabs[this.currentTabIdx]) {
+        return null
+      }
+      return this.tabs[this.currentTabIdx].name
+    }
+  },
+  methods: {
+    closeAlert() {
+      this.showAlert = false
+    },
+    formatTabName(tabContent) {
+      let name = tabContent.name ? tabContent.name : 'New API action'
+      name = truncateName(name, 11)
+      name += tabContent.saved ? '' : ' *'
+      return name
+    },
+    queryChanged({ query, tabIdx }) {
+      const savedIdx = this.tabs[tabIdx].savedIdx
+      if (savedIdx !== null && savedIdx !== undefined) {
+        this.tabs[tabIdx].saved = _.isEqual(
+          this.savedQueries[savedIdx].query,
+          query
+        )
+      }
+      this.tabs[tabIdx].query = JSON.parse(JSON.stringify(query))
+    },
+    setCurrentTab(tabIdx) {
+      this.currentTabIdx = tabIdx
+    },
+    closeTab(tabIdx) {
+      if (this.currentTabIdx === tabIdx) {
+        if (this.currentTabIdx > 0) {
+          this.currentTabIdx = tabIdx - 1
+        } else {
+          this.currentTabIdx = 0
+        }
+      } else if (this.currentTabIdx > tabIdx) {
+        this.currentTabIdx = this.currentTabIdx - 1
+      }
+      this.tabs.splice(tabIdx, 1)
+    },
+    deleteSavedQuery(savedQueryIdx) {
+      const tabIdx = this.tabs.findIndex(t => t.savedIdx === savedQueryIdx)
+      if (tabIdx !== -1) {
+        this.tabs[tabIdx].name = ''
+        this.tabs[tabIdx].savedIdx = null
+        this.tabs[tabIdx].saved = false
+      }
+      this.savedQueries.splice(savedQueryIdx, 1)
+      this.storeQueriesToLocalStorage()
+    },
+    loadSavedQuery(savedQueryIdx) {
+      if (!this.savedQueries[savedQueryIdx]) return
```

Comment:
Hmmmm not sure about how this works, maybe we better talk about it on discord

Reply:
Yes, we should improve the error handling here